app/services/completions_decider.rb

```diff
@@ -0,0 +1,34 @@
+class CompletionsDecider
+  def initialize(user_agent:, request_url:)
+    @user_agent = user_agent
+    @request_url = request_url
+  end
+
+  def continue_to_sp?
+    user_agent_matches_redirect_uri?
+  end
+
+  private
+
+  attr_reader :user_agent, :request_url
+
+  def user_agent_matches_redirect_uri?
+    desktop_and_web_redirect_uri? || mobile_and_app_redirect_uri?
```

Comment:
Good point. I didn't think of that scenario. How common is that? Do you have data that shows how many users are using the web app on a mobile device?